### PR TITLE
fix(table): Sorted/Filtered table fixed column transparent background unreadable

### LINKED ISSUE #39006: Sorted/Filtered table fixed column transparent background unreadable
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/focused-flower-zsbrg0)

### Steps to reproduce

1. Sort by the name column
2. Scroll the table to the left

### What is expected?

The content that scrolls should hide behind the fixed table column.

### What is actually happening?

The background of the fixed column is transparent which makes the content unreadable.

| Environment | Info |
| --- | --- |
| antd | 5.0.1 |
| React | 18.2.0 |
| System | MacOS |
| Browser | Firefox |

---
This is caused by the color scheme which uses  `rgba(0,0,0,0.02)` for `colorFillQuaternary` and in the header `rgba(0, 0, 0, 0.06)` for `colorFillSecondary`

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #39006 

### 💡 Background and solution

- fix: Sorted/Filtered table fixed column transparent background unreadable

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix Sorted/Filtered table fixed column transparent background unreadable |
| 🇨🇳 Chinese |  修复排序/筛选的表格的固定列背景色透明导致显示异常问题  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
随手看看 https://github.com/ant-design/ant-design/issues/38914

**Comment by kiner-tang** [comment]
> 随手看看 #38914

好的，我看看

**Comment by MadCcc** [comment]
这个透明色应该是刻意的

**Comment by MadCcc** [comment]
image
这个颜色不对了

File: components/table/style/index.tsx [hunk 1]
```diff
@@ -315,8 +314,8 @@ export default genComponentStyleHook('Table', (token) => {
     tableFooterTextColor: colorTextHeading,
     tableFooterBg: colorFillAlterSolid,
     tableHeaderCellSplitColor: colorSplit,
-    tableHeaderSortBg: colorFillSecondary,
-    tableHeaderSortHoverBg: colorFillContent,
+    tableHeaderSortBg: colorFillAlterSolid,
+    tableHeaderSortHoverBg: colorFillAlterSolid,
```
Comment (MadCcc) [inline components/table/style/index.tsx:318]: token 错了，应该分别用这些透明色算实色

File: components/table/style/index.tsx [hunk 2]
```diff
@@ -325,8 +324,8 @@ export default genComponentStyleHook('Table', (token) => {
       .clone()
       .setAlpha(baseColorActionHover.getAlpha() * opacityLoading)
       .toRgbString(),
-    tableBodySortBg: colorFillAlter,
-    tableFixedHeaderSortActiveBg: colorFillSecondary,
+    tableBodySortBg: colorFillAlterSolid,
+    tableFixedHeaderSortActiveBg: colorFillAlterSolid,
```
Comment (MadCcc) [inline components/table/style/index.tsx:328]: 这个也是

Comment (kiner-tang) [reply components/table/style/index.tsx:328]: 1

File: components/table/style/index.tsx (hunk 1 above)
Comment (kiner-tang) [reply components/table/style/index.tsx:318]: 我们有现成的方法根据透明色计算实色吗

Comment (kiner-tang) [reply components/table/style/index.tsx]: done

File: components/table/style/index.tsx (hunk 2 above)
Comment (kiner-tang) [reply components/table/style/index.tsx]: done

**Comment by kiner-tang** [comment]
@MadCcc 帮忙再看看

File: docs/react/recommendation.zh-CN.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ title: 社区精选组件
 | 路由 | [react-router](https://github.com/ReactTraining/react-router) |
 | 布局 | [react-grid-layout](https://github.com/react-grid-layout/react-grid-layout) [react-grid-system](https://github.com/sealninja/react-grid-system) [rc-dock](https://github.com/ticlo/rc-dock) |
 | 拖拽 | [dnd-kit](https://github.com/clauderic/dnd-kit) [react-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd/) [react-dnd](https://github.com/gaearon/react-dnd) |
-| 代码编辑器 | [@uiw/react-codemirror](https://github.com/uiwjs/react-codemirror) [react-monaco-editor]([https://github.com/superRaytin/react-monaco-editor](https://github.com/react-monaco-editor/react-monaco-editor)) |
+| 代码编辑器 | [@uiw/react-codemirror](https://github.com/uiwjs/react-codemirror) [react-monaco-editor](<[https://github.com/superRaytin/react-monaco-editor](https://github.com/react-monaco-editor/react-monaco-editor)>) |
```
Comment (MadCcc) [inline docs/react/recommendation.zh-CN.md:16]: 这个相关的？

File: components/table/style/index.tsx [hunk 4]
```diff
@@ -320,6 +320,13 @@ export default genComponentStyleHook('Table', (token) => {
   const tableSelectedRowBg = controlItemBgActive;
   const zIndexTableFixed: number = 2;
 
+  const colorFillSecondarySolid = new TinyColor(colorFillSecondary)
+    .onBackground(colorBgContainer)
+    .setAlpha(1);
+  const colorFillContentSolid = new TinyColor(colorFillContent)
+    .onBackground(colorBgContainer)
+    .setAlpha(1);
```
Comment (MadCcc) [inline components/table/style/index.tsx:328]:
```suggestion
  const colorFillSecondarySolid = new TinyColor(colorFillSecondary)
    .onBackground(colorBgContainer)
    .toHexString();
  const colorFillContentSolid = new TinyColor(colorFillContent)
    .onBackground(colorBgContainer)
    .toHexString();
```

File: components/table/style/index.tsx [hunk 5]
```diff
@@ -351,8 +358,8 @@ export default genComponentStyleHook('Table', (token) => {
       .clone()
       .setAlpha(baseColorActionHover.getAlpha() * opacityLoading)
       .toRgbString(),
-    tableBodySortBg: colorFillAlter,
-    tableFixedHeaderSortActiveBg: colorFillSecondary,
+    tableBodySortBg: colorFillAlterSolid,
+    tableFixedHeaderSortActiveBg: colorFillSecondarySolid.toRgbString(),
```
Comment (MadCcc) [inline components/table/style/index.tsx:362]: 直接在上面转了

File: docs/react/recommendation.zh-CN.md (hunk 3 above)
Comment (kiner-tang) [reply docs/react/recommendation.zh-CN.md:16]: 不相关的，应该是格式化了，我重置一下

Comment (kiner-tang) [reply docs/react/recommendation.zh-CN.md:16]: done

File: components/table/style/index.tsx (hunk 5 above)
Comment (kiner-tang) [reply components/table/style/index.tsx]: done

File: docs/react/recommendation.zh-CN.md (hunk 3 above)
Comment (kiner-tang) [inline docs/react/recommendation.zh-CN.md:16]: 这个应该是在`pre-commit`的时候自动格式化的。刚刚重置回来，一提交又被格式化回去了

Comment (MadCcc) [reply docs/react/recommendation.zh-CN.md:16]: prettier ignore 一下吧

Comment (kiner-tang) [reply docs/react/recommendation.zh-CN.md:16]: 好

Comment (kiner-tang) [reply docs/react/recommendation.zh-CN.md]: done

**Review by MadCcc — APPROVED** [review]

**Comment by afc163** [comment]
还有 https://github.com/ant-design/ant-design/issues/39860

**Comment by tonny008** [comment]
好像并没有修复啊，表格背景透明的情况，横向滚动固定列也会透出来后面的

**Pull request merged**